src/main/webapp/search-interview.js

```diff
@@ -54,30 +66,60 @@ function selectInterview(interviewer) {
   }
   let utcStartTime = interviewer.getAttribute('data-utc');
   let position = document.getElementById('position').value;
-  if (confirm(
-      `You selected: ${date} from ${time} with a ` +
-      `${company} ${job}. ` +
-      `Click OK if you wish to proceed.`)) {
-    alert(
-      `You have scheduled a ${position} interview on ${date}` +
-      ` from ${time} with a ${company} ` +
-      `${job}. Check your email for more ` +
-      `information.`);
-    if (company === '<Not specified>') {
-      company = '';
+  let role = selectedRole();
+  if (role === 'Interviewee') {
+    if (confirm(
+        `You selected: ${date} from ${time} with a ` +
+        `${company} ${job}. ` +
+        `Click OK if you wish to proceed.`)) {
+      alert(
+        `You have scheduled a ${position} interview on ${date}` +
+        ` from ${time} with a ${company} ` +
+        `${job}. Check your email for more ` +
+        `information.`);
+      if (company === '<Not specified>') {
+        company = '';
+      }
+      if (job === '<Not specified>') {
+        job = '';
+      }
+      let requestObject = {
+        company: company,
+        job: job,
+        utcStartTime: utcStartTime,
+        position: selectedEnumPosition()
+      };
+      let requestBody = JSON.stringify(requestObject);
+      let request = new Request('/scheduled-interviews', {method: 'POST', body: requestBody});
+      fetch(request).then(() => {window.location.replace('/scheduled-interviews.html');});
     }
-    if (job === '<Not specified>') {
-      job = '';
+  }
+  if (role === 'Shadow') {
+    if (confirm(
+        `You selected: ${date} from ${time} with a ` +
+        `${company} ${job}. ` +
+        `Click OK if you wish to proceed.`)) {
+      alert(
+        `You will shadow a ${position} interview on ${date}` +
+        ` from ${time} with a ${company} ` +
+        `${job}. Check your email for more ` +
+        `information.`);
+      if (company === '<Not specified>') {
+        company = '';
+      }
+      if (job === '<Not specified>') {
+        job = '';
+      }
+      let requestObject = {
+        company: company,
+        job: job,
+        utcStartTime: utcStartTime,
+        position: selectedEnumPosition()
+      };
+      let requestBody = JSON.stringify(requestObject);
+      let request = new Request('/scheduled-interviews', {method: 'PUT', body: requestBody});
+      fetch(request).then(() => {window.location.replace('/scheduled-interviews.html');});
```

Comment:
All this can be abstracted into an updateDatastore() method, the only change is the methodType which can be a param.

Reply:
Done